$NetBSD: patch-polyc.in,v 1.3 2015/12/29 23:34:54 dholland Exp $

Programs with C++ code need to link with the C++ compiler.
Use and honor COMPILER_RPATH_FLAG.

--- polyc.in.orig	2014-05-11 11:16:37.000000000 +0000
+++ polyc.in
@@ -2,7 +2,7 @@
 prefix=@prefix@
 exec_prefix=@exec_prefix@
 BINDIR=@bindir@
-LINK=@CC@
+LINK=@CXX@
 LIBDIR=@libdir@
 LIBS="@LIBS@ @FFI_LIBS@"
 CFLAGS="@CFLAGS@"
@@ -34,9 +34,9 @@ link()
 {
     if [ X"$2" = "X" ]
     then
-        ${LINK} ${EXTRALDFLAGS} ${CFLAGS} $1 -L${LIBDIR} -lpolymain -lpolyml ${LIBS}
+        ${LINK} ${EXTRALDFLAGS} ${CFLAGS} $1 -L${LIBDIR} @COMPILER_RPATH_FLAG@${LIBDIR} -lpolymain -lpolyml ${LIBS}
     else
-        ${LINK} ${EXTRALDFLAGS} ${CFLAGS} $1 -o $2 -L${LIBDIR} -lpolymain -lpolyml ${LIBS}
+        ${LINK} ${EXTRALDFLAGS} ${CFLAGS} $1 -o $2 -L${LIBDIR} @COMPILER_RPATH_FLAG@${LIBDIR} -lpolymain -lpolyml ${LIBS}
     fi
 }
 
